parlai/tasks/glue/agents.py

```diff
@@ -15,5 +23,73 @@ class ColaTeacher(AbstractHuggingFaceTeacher):
     hf_splits_mapping = {'train': 'train', 'valid': 'validation', 'test': 'test'}
 
 
+class MnliTeacher(AbstractHuggingFaceTeacher):
+    hf_path = 'glue'
+    hf_name = 'mnli'
+    hf_text_fields = ['premise', 'hypothesis']
+    hf_label_field = 'label'
+    hf_splits_mapping = {
+        'train': 'train',
+        'valid': 'validation_matched',
+        'test': 'test_matched',
+    }
```

Comment:
Mnli dataset also has `validation_mismatched` and `test_mismatched` splits, I just picked one for the teacher.

Reply:
Hm, can you go read the superglue and mnli paper and try to figure out which is right? We shouldn't guess.

Reply:
"For MNLI, we report accuracy averaged over the matched and mismatched test sets" ... they use both.